src/runtime/pythonengine.cs

```diff
@@ -64,7 +67,17 @@ public static string ProgramName
 
                 return result ?? "";
             }
-            set { Runtime.Py_SetProgramName(value); }
+            set
+            {
+                if (_programName != IntPtr.Zero)
```

Comment:
according to docs this is fine: https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.marshal.freehglobal(v=vs.110).aspx#Anchor_1